### DISCUSSION TITLE
Move extension publisher without changing extension id

### DISCUSSION THREAD

**Discussion by Ikuyadeu** [microsoft/vsmarketplace#142] [Feedback]
Hello, Marketplace team.
Thank you for give us a great platform.

I have just sent you a request for my project [vscode-r](https://marketplace.visualstudio.com/items?itemName=Ikuyadeu.r) migration.
Currently, my project is copyrighted to `REditorSupport`  because other members mainly maintain it.
I'm going to change my display name on the marketplace.

I have four options, but 2 good solution needs to Marketplace member help.

* Best: I would like to change the display name of the Marketplace to `REditorSupport` without changing the publisher id.
* Good: Move to REditor Support by change the publisher id `ikuyadeu.r` to `reditorsupport.r`. However, in that case, it will affect those who have dependencies on our extension.
* Bad: Publish as the new extension.
* Bad: Keep current state

I would be happy if you help me to choose the best option.

Regards

CC @renkun-ken

**Comment by prashantvc** [comment]
I am sorry, this is somehow slipped through the cracks.

Best option is the easiest. All you have to do is update the `displayName` in the `package.json` file to `REditorSupport`; then upload the new package.

The _Good_ option is not so good; you may lose few older versions if we change the publisher's name

**Reply by renkun-ken** [reply]
But the `displayName` in `project.json` seems to mean the display name of the extension rather than the publisher. This is the current value:

<img width="476" alt="image" src="https://user-images.githubusercontent.com/4662568/161510989-80c2e03b-b814-4245-a606-329269220d26.png">
